app/src/features/sources/SourceCard.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+import { Source } from "services/api/generated/api.generated";
+import { useTranslation } from "react-i18next";
+import { useDestroySourceMutation } from "services/api/api";
+import clsx from "clsx";
+
+import {
+  Card,
+  CardContent,
+  CardHeader,
+  Divider,
+  IconButton,
+  ListItemIcon,
+  ListItemText,
+  makeStyles,
+  Menu,
+  MenuItem,
+  Typography,
+} from "@material-ui/core";
+
+import { ReactComponent as Mapping } from "assets/icons/mapping_icon.svg";
+import More from "@material-ui/icons/MoreHoriz";
+import ManagePermissions from "@material-ui/icons/Person";
+import Delete from "@material-ui/icons/Delete";
+import Edit from "@material-ui/icons/Edit";
+import Attribute from "@material-ui/icons/LocalOffer";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: 300,
+    textAlign: "left",
+  },
+  listItemIcon: {
+    minWidth: 33,
+    color: theme.palette.text.primary,
+  },
+  delete: {
+    color: theme.palette.error.light,
+  },
+  actionButton: {
+    border: `1px solid ${
+      theme.palette.type === "dark"
+        ? theme.palette.grey[600]
+        : theme.palette.grey[300]
+    }`,
+    borderRadius: 5,
+  },
+  icon: {
+    width: 12,
+    height: 12,
+    marginRight: theme.spacing(1),
+    "& path": {
+      fill: theme.palette.text.secondary,
+    },
+  },
+  sourceDetail: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+  },
+}));
+
+type SourceCardProps = {
+  source: Source;
+  mappingCount?: number;
+  attributesCount?: number;
+  onClick?: (id?: string) => void;
+  editSource?: (source: Source) => void;
+};
+
+const SourceCard = ({
+  source,
+  mappingCount,
+  attributesCount,
+  onClick,
+  editSource,
+}: SourceCardProps): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const [deleteSource] = useDestroySourceMutation();
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const _handleClick = () => {
+    onClick && onClick(source.id);
+  };
+  const _handleMenuClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const _handleClose = () => {
+    setAnchorEl(null);
+  };
+  const _managePermissions = () => {
+    _handleClose();
+  };
+  const _rename = () => {
+    editSource && editSource(source);
```

Comment:
Here we can just call `usePartialUpdateSourceMutation`. The list will update automatically. No need to pass editSource all the way down here or even create it.